lib/iris/pandas.py

```diff
@@ -192,39 +134,42 @@ def as_data_frame(cube, copy=True):
 
     Kwargs:
 
-        * copy - Whether to make a copy of the data.
-                 Defaults to True. Must be True for masked data
-                 and some data types (see notes below).
+        * dropna - Remove missing values from returned dataframe.
+                    Defaults to True.
 
     .. note::
 
-        This function will copy your data by default.
-        If you have a large array that cannot be copied,
-        make sure it is not masked and use copy=False.
-
-    .. note::
-
-        Pandas will sometimes make a copy of the array,
-        for example when creating from an int32 array.
-        Iris will detect this and raise an exception if copy=False.
+        TBC
 
     """
     data = cube.data
     if ma.isMaskedArray(data):
-        if not copy:
-            raise ValueError("Masked arrays must always be copied.")
         data = data.astype("f").filled(np.nan)
     elif copy:
         data = data.copy()
 
-    index = columns = None
-    if cube.coords(dimensions=[0]):
-        index = _as_pandas_coord(cube.coord(dimensions=[0]))
-    if cube.coords(dimensions=[1]):
-        columns = _as_pandas_coord(cube.coord(dimensions=[1]))
+    # Extract dim coord information
+    if cube.ndim != len(cube.dim_coords):
+        # Create dummy dim coord information if dim coords not defined
+        coord_names = ["dim" + str(n) for n in range(cube.ndim)]
+        coords = [range(dim) for dim in cube.shape]
+        for c in cube.dim_coords:
+            for i, dummyc in enumerate(coords):
+                if len(dummyc) == len(c.points):
+                    coords[i] = _as_pandas_coord(c)
+                    coord_names[i] = c.name()
+                else:
+                    pass
+    else:
+        coord_names = list(map(lambda x: x.name(), cube.dim_coords))
+        coords = list(map(lambda x: _as_pandas_coord(x), cube.dim_coords))
+
+    index = pandas.MultiIndex.from_product(coords, names=coord_names)
+    data_frame = pandas.DataFrame({cube.name(): data.flatten()}, index=index)
```

Comment:
```suggestion
    data_frame = pandas.DataFrame(data.ravel(), columns=[cube.name()], index=index)
```

This allows `data_frame` to directly use `cube.data` i.e. `copy=False`.